react-carousel/my-app/src/Carousel.tsx

```diff
@@ -0,0 +1,44 @@
+import { NextButton } from './NextButton';
+import { PrevButton } from './PrevButton';
+import { Dots } from './Dots';
+import { ImageCard } from './ImageCard';
+import { images } from './App';
+import { useEffect, useState } from 'react';
+type images = {
+  src: string;
+  alt: string;
+};
+type Props = {
+  images: images[];
+};
+
+export function Carousel({ images }: Props) {
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  useEffect(() => {
+    const timerId = setTimeout(() => {
+      setCurrentIndex((currentIndex) => (currentIndex + 1) % images.length);
+    }, 2000);
+    return () => clearTimeout(timerId);
+  }, [currentIndex, images.length]);
+
+  function handleNextClick() {
+    setCurrentIndex((prev) => (prev + 1) % images.length);
```

Comment:
In these ones, you can actually just use the simpler `setCurrentIndex((currentIndex + 1) % images.length);`.

I usually use this simpler form first, and then only switch to the more complex form with an arrow function only if problems arise because of closures.